### PR TITLE
UPC-43 Make changelogUrl and downloadUrl URLs

### PR DESCRIPTION
**Pull request by erichirlemann**

### CODE REVIEW

File: sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/UpdateCenterDeserializer.java [hunk 1]
```diff
@@ -235,22 +235,27 @@ private void parsePluginReleases(Properties p, Sonar sonar, String pluginKey, Pl
     }
   }
 
-  private Release parsePluginRelease(Properties p, Sonar sonar, String pluginKey, Plugin plugin, boolean isPublicRelease, boolean isArchivedRelease, String pluginVersion) {
+  private Release parsePluginRelease(Properties p, Sonar sonar, String pluginKey, Plugin plugin, boolean isPublicRelease, boolean isArchivedRelease, String pluginVersion) throws IllegalArgumentException {
```
Comment (SonarTech) [inline sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/UpdateCenterDeserializer.java]: MAJOR Remove the declaration of thrown exception 'java.lang.IllegalArgumentException' which is a runtime exception. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
MINOR Split this 204 characters long line (which is greater than 180 authorized). [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS00103)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/UpdateCenterDeserializer.java [hunk 2]
```diff
@@ -235,7 +235,13 @@ private void parsePluginReleases(Properties p, Sonar sonar, String pluginKey, Pl
     }
   }
 
-  private Release parsePluginRelease(Properties p, Sonar sonar, String pluginKey, Plugin plugin, boolean isPublicRelease, boolean isArchivedRelease, String pluginVersion) throws IllegalArgumentException {
+  private Release parsePluginRelease(Properties p
+                                    , Sonar sonar
```
Comment (henryju) [inline sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/UpdateCenterDeserializer.java]: Seems strange to me to have comma upfront. Not sure our coding conventions are precise enough on this topic, but that's the first time I see that in our codebase.

File: sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/UpdateCenterDeserializer.java [hunk 3]
```diff
@@ -235,22 +235,33 @@ private void parsePluginReleases(Properties p, Sonar sonar, String pluginKey, Pl
     }
   }
 
-  private Release parsePluginRelease(Properties p, Sonar sonar, String pluginKey, Plugin plugin, boolean isPublicRelease, boolean isArchivedRelease, String pluginVersion) {
+  private Release parsePluginRelease(Properties p
+      , Sonar sonar
```
Comment (henryju) [inline sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/UpdateCenterDeserializer.java]: Not sure my previous comment was preserved so I'm repeating here: having comma upfront looks strange. Our coding conventions are probably not precise enough but that the first time I see that in our codebase.

**Comment by henryju** [comment]
Minor feedback about code style but then LGTM

File: sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/UpdateCenterDeserializer.java (hunk 3 above)
Comment (erichirlemann) [inline sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/UpdateCenterDeserializer.java]: True, the usage is rather to comma+CR.
Fixed.

**Pull request merged**